core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -561,7 +561,9 @@ public void start(Boolean isPrimary) throws IOException {
         MountInfo mountInfo = mMountTable.getMountTable().get(key);
         UnderFileSystemConfiguration ufsConf =
             UnderFileSystemConfiguration.defaults(ServerConfiguration.global())
-            .createMountSpecificConf(mountInfo.getOptions().getPropertiesMap());
+                .createMountSpecificConf(mountInfo.getOptions().getPropertiesMap())
```

Comment:
The API here is pretty confusing, it is not obvious we need to set the read only or shared fields separately.
For the fix to released branches I'm fine with this, to master we should should consider something else. How about a `fromMountInfo` method?

Reply:
makes sense. not targeting this PR though as its a big refactor

Reply:
Can we file a tech-debt issue for this?

Reply:
done